bundles/persistence/org.openhab.persistence.influxdb/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -415,24 +421,31 @@ private Object convertBigDecimalToNum(BigDecimal value) {
   /**
    * Converts {@link State} to objects fitting into influxdb values.
    * 
-   * @param state to be converted
-   * @return integer or double value for DecimalType and PercentType, an integer for DateTimeType
-   *         and 0 or 1 for OnOffType and OpenClosedType.
+   * @param 	state to be converted
+   * @return 	integer or double value for DecimalType and PercentType, 
+   * 			0 or 1 for OnOffType and OpenClosedType,
+   *         	SortedMap<String, PrimitiveType> for HSBType,
+   *         	integer for DateTimeType,
+   *         	SortedMap<String, PrimitiveType> for PointType
    */
   private Object stateToObject(State state) {
     Object value;
-    if (state instanceof PercentType) {
-      value = convertBigDecimalToNum(((PercentType) state).toBigDecimal());
-    } else if (state instanceof DecimalType) {
-      value = convertBigDecimalToNum(((DecimalType) state).toBigDecimal());
-    } else if (state instanceof DateTimeType) {
-      value = ((DateTimeType) state).getCalendar().getTime().getTime();
+    if (state instanceof DecimalType) {
+    	value = convertBigDecimalToNum(((DecimalType) state).toBigDecimal());
+    } else if (state instanceof PercentType) {
+        value = convertBigDecimalToNum(((PercentType) state).toBigDecimal());
     } else if (state instanceof OnOffType) {
-      value = (OnOffType) state == OnOffType.ON ? 1 : 0;
+    	value = (OnOffType) state == OnOffType.ON ? 1 : 0;
     } else if (state instanceof OpenClosedType) {
-      value = (OpenClosedType) state == OpenClosedType.OPEN ? 1 : 0;
+    	value = (OpenClosedType) state == OpenClosedType.OPEN ? 1 : 0;
+    } else if (state instanceof HSBType) {
+    	value = ((HSBType) state).getConstituents();
+    } else if (state instanceof DateTimeType) {
+    	value = ((DateTimeType) state).getCalendar().getTime().getTime();
+    } else if (state instanceof PointType) {
+    	value = ((PointType) state).getConstituents();
```

Comment:
We need a converter for PointType to String. Unfortunately ((PointType) state).toString() is odd because we can not feed this value back into a PointType constructor. I think a concatenated comma separated String of latitude, longitude, altitude should work. On restore this string has to be converted to a valid input for a PointType constructor. A similar approach can be found in HSBType constructor HSBType(String value).